src/main/java/org/opensearch/neuralsearch/processor/TextChunkingProcessor.java

```diff
@@ -149,14 +145,15 @@ private boolean isListOfString(final Object value) {
     }
 
     private int getMaxTokenCount(final Map<String, Object> sourceAndMetadataMap) {
+        int defaultMaxTokenCount = IndexSettings.MAX_TOKEN_COUNT_SETTING.get(environment.settings());
         String indexName = sourceAndMetadataMap.get(IndexFieldMapper.NAME).toString();
         IndexMetadata indexMetadata = clusterService.state().metadata().index(indexName);
         if (Objects.isNull(indexMetadata)) {
-            return IndexSettings.MAX_TOKEN_COUNT_SETTING.get(environment.settings());
+            return defaultMaxTokenCount;
         }
         // if the index is specified in the metadata, read maxTokenCount from the index setting
-        IndexService indexService = indicesService.indexServiceSafe(indexMetadata.getIndex());
-        return indexService.getIndexSettings().getMaxTokenCount();
+        String maxTokenCountString = indexMetadata.getSettings().get("index.analyze.max_token_count");
+        return NumberUtils.toInt(maxTokenCountString, defaultMaxTokenCount);
```

Comment:
```suggestion
        return IndexSettings.MAX_TOKEN_COUNT_SETTING.getDefault(indexMetadata.getSettings());
```

Reply:
Already updated

Reply:
It takes me a while to test this recommended change.

Reply:
Unfortunately, this code change always return the default value for max token count.

Reply:
Can we check if there's any other method like `getDefault` we can use here? I don't think using explicit string `index.analyze.max_token_count` is a good practice, better to use the enum `IndexSettings.MAX_TOKEN_COUNT_SETTING`.

Reply:
Already updated